### PR TITLE
Adds binary 1.1 read support for symbol address annotations

### PR DESCRIPTION
**Pull request by zslayton**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/lazy/binary/raw/v1_1/immutable_buffer.rs [hunk 1]
```diff
@@ -505,8 +505,7 @@ impl<'a> ImmutableBuffer<'a> {
             }
             Ok((None, after_nops))
         } else {
-            self.read_value(opcode)
-                .map(|(v, after)| (Some(v), after))
+            self.read_value(opcode).map(|(v, after)| (Some(v), after))
```
Comment (zslayton) [inline src/lazy/binary/raw/v1_1/immutable_buffer.rs:508]: 🗺️ `cargo fmt` re-flowed some things in this file even though they hadn't changed in this commit. You can ignore this and the one below (lines 630-654).

File: src/lazy/binary/raw/v1_1/immutable_buffer.rs [hunk 2]
```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
         opcode: Opcode,
     ) -> ParseResult<'a, EncodedAnnotations> {
         let input_after_opcode = self.consume(1);
-        // TODO: This implementation actively reads the annotations, which isn't necessary.
-        //       At this phase of parsing we can just identify the buffer slice that contains
-        //       the annotations and remember their encoding; later on, the annotations iterator
-        //       can actually do the reading. That optimization would be impactful for FlexSyms
-        //       that represent inline text.
```
Comment (zslayton) [inline src/lazy/binary/raw/v1_1/immutable_buffer.rs:735]: 🗺️ I did this optimization for skipping `FlexUInt` symbol addresses, but doing it for `FlexSym` would be much more involved. I introduced two methods below to encapsulate that logic:
* `consume_flex_uint` (skips without reading)
* `consume_flex_sym` (skips by reading)

File: src/lazy/binary/raw/v1_1/immutable_buffer.rs [hunk 3]
```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
...
-                let (flex_sym, remaining_input) = input_after_opcode.read_flex_sym()?;
-                let sequence = EncodedAnnotations {
-                    encoding: AnnotationsEncoding::FlexSym,
-                    header_length: 1, // 0xE7
-                    sequence_length: u16::try_from(flex_sym.size_in_bytes()).map_err(|_| {
-                        IonError::decoding_error(
-                            "the maximum supported annotations sequence length is 65KB.",
-                        )
-                    })?,
-                };
```
Comment (zslayton) [inline src/lazy/binary/raw/v1_1/immutable_buffer.rs:747]: 🗺️ Previously I was constructing an `EncodedAnnotations` struct in each `match` arm. I lifted out out to the end of the method, which I think helped readability.

File: src/lazy/binary/raw/v1_1/immutable_buffer.rs [hunk 4]
```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
...
+                    IonError::decoding_error(
+                        "the maximum supported annotations sequence length is 65KB",
+                    )
+                })?;
+
+        let sequence = EncodedAnnotations {
+            encoding: AnnotationsEncoding::FlexSym,
+            header_length,
+            sequence_length,
+        };
```
Comment (zslayton) [inline src/lazy/binary/raw/v1_1/immutable_buffer.rs:781]: 🗺️ Here's the instantiation of the `EncodedAnnotations`.

File: src/lazy/binary/raw/v1_1/immutable_buffer.rs [hunk 5]
```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
...
+
         Ok((sequence, remaining_input))
     }
 
+    /// Skips beyond a `FlexSym` at the head of the buffer, returning the remaining slice.
+    fn consume_flex_sym(self) -> IonResult<Self> {
+        // TODO: As an optimization, see if we can avoid actually reading the flex_sym.
+        let (flex_sym, remaining) = self.read_flex_sym()?;
+        if let FlexSymValue::Opcode(opcode) = flex_sym.value() {
+            todo!("FlexSym escapes in annotation sequences; opcode: {opcode:?}");
```
Comment (zslayton) [inline src/lazy/binary/raw/v1_1/immutable_buffer.rs:791]: 🗺️ The skipper doesn't (yet) attempt to interpret the opcode and skip whatever it indicates.

File: src/lazy/binary/raw/v1_1/immutable_buffer.rs [hunk 6]
```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
...
+        }
+
+        Ok(remaining)
+    }
+
+    /// Skips beyond a `FlexUInt` at the head of the buffer, returning the remaining slice.
+    fn consume_flex_uint(self) -> IonResult<Self> {
```
Comment (zslayton) [inline src/lazy/binary/raw/v1_1/immutable_buffer.rs:798]: 🗺️ Skips a `FlexUInt` without actually reading it.

**Review by zslayton — COMMENTED** [review]
🗺️ PR tour

**Review by popematt — APPROVED** [review]

**Pull request merged**